client/question/components/SpeechBalloonsContainerDirective.js

```diff
@@ -50,147 +43,143 @@ tie.directive('speechBalloonsContainer', [function() {
                     </output-snippet>
                   </span>
                 </p>
-              </div>
-              <div class="tie-left-speech-balloon-tail">
-                <div class="tie-speech-balloon-tail-outer"></div>
-                <div class="tie-speech-balloon-tail-inner"></div>
-              </div>
+              </tie-speech-balloon-left>
+              <tie-speech-balloon-tail-left></tie-speech-balloon-tail-left>
             </div>
-
             <div ng-if="!balloon.isDisplayedOnLeft()">
-              <div class="tie-speech-balloon tie-right-speech-balloon">
+              <tie-speech-balloon-right>
                 <p ng-repeat="paragraph in balloon.getFeedbackParagraphs() track by $index"
-                   class="tie-feedback-paragraph tie-feedback-paragraph-code">
+                    class="tie-feedback-paragraph tie-feedback-paragraph-code">
                   <code-snippet content="paragraph.getContent()"></code-snippet>
                 </p>
-              </div>
-              <div class="tie-right-speech-balloon-tail">
-                <div class="tie-speech-balloon-tail-outer"></div>
-                <div class="tie-speech-balloon-tail-inner"></div>
-              </div>
+              </tie-speech-balloon-right>
+              <tie-speech-balloon-tail-right></tie-speech-balloon-tail-right>
             </div>
-          </div>
+          </tie-speech-balloon-container>
         </div>
       </div>
 
       <style>
+        .tie-feedback-paragraph {
```

Comment:
For the paragraphs, I feel like there might be a bit too much extra padding on the inside between the bubble and the text? You'll be able to see more of the feedback I think if you decreased the spacing. Might just be my own personal preference though.

Reply:
Reduced padding on top and bottom as suggested.